test/src/tests/generated/body-complex/models.go

```diff
@@ -141,7 +141,9 @@ func (c Cookiecuttershark) MarshalJSON() ([]byte, error) {
 	if c.Siblings != nil {
 		objectMap["siblings"] = c.Siblings
 	}
-	objectMap["fishtype"] = c.Fishtype
+	if c.Fishtype != "" {
+		objectMap["fishtype"] = c.Fishtype
```

Comment:
hmm so I guess this is a valid case ? Also I was thinking if it impacts polymorphic types since they should always have a value

Reply:
acutally for the discriminator value we always assign it so that should be fine. So when is hte enum a null value ? when its missing entirely ? ( is optional )

Reply:
Good question.  Presumably for the discriminator it should never be empty?

Reply:
yes and it will never be empty because we assign with the value for the type  in the beginning of the method

Reply:
Yep optional fields that are enum types could have an empty value.